### PR TITLE
Leveldown instance as parameter

### PR DESCRIPTION
**Pull request by ralphtheninja**
Closes https://github.com/Level/levelup/issues/438 https://github.com/Level/levelup/issues/443

### CODE REVIEW

File: lib/levelup.js [hunk 1]
```diff
@@ -62,14 +61,17 @@ function LevelUP (location, options, callback) {
...
 
-  if ((!options || typeof options.db !== 'function') && typeof location !== 'string') {
-    error = new InitializationError(
-        'Must provide a location for the database')
-    if (callback) {
-      return process.nextTick(function () {
-        callback(error)
-      })
-    }
+  if (!options || typeof options.db !== 'function') {
```
Comment (ralphtheninja) [inline lib/levelup.js]: @juliangruber If we're happy with this PR we might want to consider squashing this commit with the next. It was easier for me to follow along what you did on `v2` branch before removing `location` completely.

File: package.json [hunk 2]
```diff
@@ -61,10 +60,8 @@
     "tape": "^4.7.0"
   },
   "browser": {
-    "./lib/leveldown.js": false,
     "leveldown": false,
-    "leveldown/package": false,
-    "semver": false
+    "leveldown/package": false
```
Comment (ralphtheninja) [inline package.json]: Are these really needed now? It seems to me we can remove the whole `browser` object unless there's something I'm missing.

**Comment by ralphtheninja** [comment]
@juliangruber If we're happy with this PR we might want to consider squashing the first and second commits into one. It was easier for me to follow along what you did on v2 branch before removing location completely, but most of the first commit is overwritten by the second one.

File: package.json (hunk 2 above)
Comment (juliangruber) [inline package.json]: yeah! seems like we can just remove the whole thing :)

**Comment by juliangruber** [comment]
aw yes!!!!

**Pull request merged**